File: C#/TestCYBRCCP.cs
using System;
using System.IO;
using System.Net;
using System.Security.Cryptography.X509Certificates;
using System.Text;

namespace TestCybrCCP
{
    class Program
    {
        static void Main(string[] args)
        {
            try
            {
                Console.WriteLine("CyberArk CCP Testing tool. Writen by Huy.Do@CyberArk.com");
                if (args.Length != 0 && args.Length != 3) {
                    Console.WriteLine("Parameters error. Please run this tool with below syntax:");
                    Console.WriteLine("TestCybrCCP \"<ccpQueryString>\" <certPfxFile> <certPassword>");
                    Console.WriteLine("Example:");
                    Console.WriteLine(".\\TestCybrCCP \"https://172.16.100.21/AIMWebService/api/Accounts?AppID=TestingCP_UAT&Safe=TestingCP_Safe&Folder=Root&Object=testcp01\" C:\\cybr\\clients.cybr.huydo.net.pfx ChangeMe123!");

                    Console.ReadLine();
                    return;
                }
                var ccpQueryString = "https://172.16.100.21/AIMWebService/api/Accounts?AppID=TestingCP_UAT&Safe=TestingCP_Safe&Folder=Root&Object=testcp01";
                var certFile = "C:\\cybr\\clients.cybr.huydo.net.pfx";
                var password = "ChangeMe123!";

                //var certFile = "C:\\cybr\\pimtest.pfx";
                //var password = "root";
                if (args.Length == 3) {
                    ccpQueryString = args[0];
                    certFile = args[1];
                    password = args[2];
                }

                Console.WriteLine("Running parameters... ");
                Console.WriteLine("ccpQueryString: " + ccpQueryString);
                Console.WriteLine("certPfxFile: " + certFile);
                Console.WriteLine("certPassword: " + password);

                var cert = new X509Certificate2(certFile, password, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.DefaultKeySet);
                var certCollection = new X509Certificate2Collection();
                certCollection.Add(cert);

                var urlRequest = ccpQueryString;
                var req = (HttpWebRequest)WebRequest.Create(urlRequest);

                ServicePointManager.ServerCertificateValidationCallback += (sender, certificate, chain, sslPolicyErrors) => true; //untrust ssl
                ServicePointManager.Expect100Continue = true;
                ServicePointManager.DefaultConnectionLimit = 9999;
                ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls;

                req.ProtocolVersion = HttpVersion.Version11;
                req.Accept = "*";
                req.Method = "GET";
                req.KeepAlive = true;
                req.ClientCertificates = certCollection;

                HttpWebResponse response = (HttpWebResponse)req.GetResponse();

                var encoding = Encoding.ASCII;
                using (var reader = new StreamReader(response.GetResponseStream(), encoding))
                {
                    string responseText = reader.ReadToEnd();
                    Console.WriteLine("Response: " + responseText);
                }
            }
            catch (System.Exception ex)
            {
                Console.WriteLine("Get Password Error: " + ex.Message);
            }
            Console.WriteLine("Press any key to exit...");
            Console.ReadLine();
        }
    }
}

File: Python/TestCYBRCCP.py
import http.client
import ssl
import json

ccp_host = '172.16.100.21'
ccp_api = '/AIMWebService/api/Accounts?AppID=TestCCP-HUYDO&Safe=TestingCP_Safe&Folder=Root&Object=testcp01'

#context = ssl.create_default_context()
#Using code below to load https CA
#context.load_verify_locations(cafile='C:\certs\cachain.crt')

#Using code below to disable https cert validation
context = ssl._create_unverified_context()
#Using code below to load client certificate for http request
context.load_cert_chain('../test/clients.cybr.huydo.net.crt', '../test/clients.cybr.huydo.net.key')

try:
    conn = http.client.HTTPSConnection(ccp_host, context = context)
    headers = {'Content-type': 'application/json'}
    conn.request('GET', ccp_api, '', headers)
    res = conn.getresponse()
    data = res.read()
    status_code = res.status
    conn.close()

    if status_code != 200:
        raise Exception('ERROR: Status Code: {}'.format(status_code))
 
# Capture Any Exceptions that Occur
except Exception as e:
    # Print Exception Details and Exit
    print(Exception(e))
    exit()
        
# Deal with Python dict for return variable
result = json.loads(data.decode('UTF-8'))

print('Username: {}'.format(result['UserName']))
print('Password: {}'.format(result['Content']))

